run_dir/static/js/project_samples.js

```diff
@@ -352,36 +375,59 @@ function load_all_udfs(){
         }
       }
 
-      // Make the project emails clickable, and add labels.
+      // Make the project emails clickable and add labels.
       else if (prettify(key) == 'contact'){
         function elabel(text, label) {
           return '<span class="label label-'+label+'">'+text+'</span>'
         }
-        if(data['order_details']) {
+        function validateEmail(email) {
+          var cap_email = null;
+          // Stolen, without shame, from https://stackoverflow.com/a/46181
+          var re = /(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))/;
+          if (email){
+               matches = email.match(re);
+               if (matches){
+                  cap_email = matches[0].toLowerCase();
+               }
+          }
+          return cap_email;
+        }
+
+        var email_html = '';
+        try {
           var emails = {}
-          var contact = data['order_details']['owner']['email'];
+          var contact = validateEmail(data['order_details']['owner']['email']);
           emails[contact] = [elabel('Contact', 'info')];
-          var lab = data['order_details']['fields']['project_lab_email'];
+          var lab = validateEmail(data['order_details']['fields']['project_lab_email']);
           if(!emails[lab]){emails[lab]=[elabel('Lab', 'default')]} else{emails[lab].push(elabel('Lab', 'default'))};
-          var bx = data['order_details']['fields']['project_bx_email'];
+          var bx = validateEmail(data['order_details']['fields']['project_bx_email']);
           if(!emails[bx]){emails[bx]=[elabel('Bioinfo', 'default')]} else {emails[bx].push(elabel('Bioinfo', 'default'))};
-          var pi = data['order_details']['fields']['project_pi_email'];
+          var pi = validateEmail(data['order_details']['fields']['project_pi_email']);
           if(!emails[pi]){emails[pi]=[elabel('PI', 'default')]} else {emails[pi].push(elabel('PI', 'default'))};
 
-          email_html = '';
           Object.keys(emails).forEach(function(k, i) {
-            if (k != 'null'){ // A bit ugly ¯\_(ツ)_/¯
+            if (k != 'null' && !validateEmail(k)) {
```

Comment:
@remiolsen  Will there be any case to enter this `if`. Cause all the 'k' of `emails` hash were by-product of `validateEmail` to begin with right. So isn't this kinda redundant ?

Reply:
No. It might also be `"null"` because it was `"null"` in the database 😓(hence my comment about this being ugly)
It's a bit strange. I mean it was all json to begin with. My thought: If it works, it works :wink: